contracts/IbbtcVaultZap.sol

```diff
@@ -197,11 +207,37 @@ contract IbbtcVaultZap is PausableUpgradeable {
         return _vaultShares(crvLp);
     }
 
+    function expectedAmount(uint256[4] calldata _amounts, bool _mintIbbtc)
+        public
+        view
+        returns (uint256 amount)
+    {
+        uint256[4] memory depositAmounts = _constructDeposit(
+            _amounts,
+            _mintIbbtc
+        );
+
+        ERC20Upgradeable[4] memory assets = [
+            ERC20Upgradeable(address(IBBTC)),
+            ERC20Upgradeable(address(RENBTC)),
+            ERC20Upgradeable(address(WBTC)),
+            ERC20Upgradeable(address(SBTC))
+        ];
+        uint256 virtualPrice = CURVE_REN_POOL.get_virtual_price();
+        for (uint256 i = 0; i < 4; i++) {
+            amount = amount.add(
+                depositAmounts[i].mul(1e36).div(virtualPrice).div(
+                    10**uint256(assets[i].decimals())
+                )
+            );
```

Comment:
Can divide by virtual price outside the loop as well to prevent rounding errors

Reply:
done